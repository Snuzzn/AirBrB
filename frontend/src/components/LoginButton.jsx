import React from 'react';
import { AiOutlineArrowRight } from 'react-icons/ai';
import PropTypes from 'prop-types';

const LoginButton = ({ handleClick, buttonText = 'Submit', label }) => {
  return (
    <button
    type="submit"
    className="items-center flex justify-center w-full px-4 py-2 text-sm bg-red-300 border border-transparent rounded-lg hover:bg-red-500 focus:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 focus-visible:ring-blue-500"
    onClick={handleClick}
    >
      <div className="flex-1">
        {buttonText}
      </div>
      <AiOutlineArrowRight className="text-lg" aria-label="Submit login icon: arrow moving right" />
    </button>
  )
}

export default LoginButton;

LoginButton.propTypes = {
  handleClick: PropTypes.func,
  buttonText: PropTypes.string,
}
