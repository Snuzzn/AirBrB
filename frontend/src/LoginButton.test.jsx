import React from 'react';
import LoginButton from './components/LoginButton';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { AiOutlineArrowRight } from 'react-icons/ai';

describe('LoginButton', () => {
  /**
   * LoginButton takes in a click function, optional text, and
   * optional label to display. It is used across multiple components,
   * including for login and registration.
   */

  const noop = () => {};

  it('triggers onClick event handler when clicked', () => {
    const onClick = jest.fn();
    shallow(<LoginButton handleClick={onClick} />).simulate('click');
    expect(onClick).toHaveBeenCalledTimes(1);
  })

  it('uses custom title', () => {
    const customText = 'custom text';
    const button = shallow(<LoginButton
      onClick={noop}
      buttonText={customText}
    />);
    // Button contains one div to hold the custom text
    expect(button.find('div').text()).toBe(customText);
  })

  it('uses default title if no custom text provided', () => {
    const button = shallow(<LoginButton
      onClick={noop}
    />);
    // Button's default text is 'Submit'
    expect(button.find('div').text()).toBe('Submit');
  })

  it('contains icon indicating submission', () => {
    const button = shallow(<LoginButton
      onClick={noop}
    />);
    expect(button.contains(<AiOutlineArrowRight />))
  })

  it('button has aria-label prop for accessibility', () => {
    const button = shallow(<LoginButton
      onClick={noop}
      label={'test'}
    />);
    expect(button.prop('aria-label')).toBe('test');
  })

  it('button uses default label if no custom label provided', () => {
    const button = shallow(<LoginButton
      onClick={noop}
    />);
    expect(button.prop('aria-label')).toBe('Submit');
  })

  // Snapshots
  it('renders with minimal props', () => {
    const button = renderer.create(<LoginButton
      onClick={noop}
    />).toJSON();
    expect(button).toMatchSnapshot();
  })

  it('renders with provided text', () => {
    const button = renderer.create(<LoginButton
      onClick={noop}
      type='I am a button!'
    />).toJSON();
    expect(button).toMatchSnapshot();
  })
})
