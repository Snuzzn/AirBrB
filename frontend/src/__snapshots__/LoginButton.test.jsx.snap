// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginButton renders with minimal props 1`] = `
<button
  aria-label="Submit"
  className="items-center flex justify-center w-full px-4 py-2 text-sm bg-red-300 border border-transparent rounded-lg hover:bg-red-500 focus:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 focus-visible:ring-blue-500"
  type="submit"
>
  <div
    className="flex-1"
  >
    Submit
  </div>
  <svg
    className="text-lg"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 1024 1024"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M869 487.8L491.2 159.9c-2.9-2.5-6.6-3.9-10.5-3.9h-88.5c-7.4 0-10.8 9.2-5.2 14l350.2 304H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h585.1L386.9 854c-5.6 4.9-2.2 14 5.2 14h91.5c1.9 0 3.8-.7 5.2-2L869 536.2a32.07 32.07 0 0 0 0-48.4z"
    />
  </svg>
</button>
`;

exports[`LoginButton renders with provided text 1`] = `
<button
  aria-label="Submit"
  className="items-center flex justify-center w-full px-4 py-2 text-sm bg-red-300 border border-transparent rounded-lg hover:bg-red-500 focus:outline-none focus-visible:ring-2 focus-visible:ring-offset-2 focus-visible:ring-blue-500"
  type="submit"
>
  <div
    className="flex-1"
  >
    Submit
  </div>
  <svg
    className="text-lg"
    fill="currentColor"
    height="1em"
    stroke="currentColor"
    strokeWidth="0"
    style={
      Object {
        "color": undefined,
      }
    }
    viewBox="0 0 1024 1024"
    width="1em"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M869 487.8L491.2 159.9c-2.9-2.5-6.6-3.9-10.5-3.9h-88.5c-7.4 0-10.8 9.2-5.2 14l350.2 304H152c-4.4 0-8 3.6-8 8v60c0 4.4 3.6 8 8 8h585.1L386.9 854c-5.6 4.9-2.2 14 5.2 14h91.5c1.9 0 3.8-.7 5.2-2L869 536.2a32.07 32.07 0 0 0 0-48.4z"
    />
  </svg>
</button>
`;
